lib/ansible/executor/module_common.py

```diff
@@ -1290,6 +1292,7 @@ def modify_module(
         become_plugin=become_plugin,
         environment=environment,
         remote_is_local=remote_is_local,
+        module_temp_dir=module_temp_dir,
```

Comment:
For this to work properly, you have to check for `if module_temp_dir is None:` before passing it into the function that you declared expects to accept a `str`.